lib/irc/IrcServer.js

```diff
@@ -6,6 +6,8 @@ var logging = require("../logging");
 var IrcClientConfig = require("../models/IrcClientConfig");
 var log = logging.get("IrcServer");
 
+const GROUP_ID_REGEX = /^\+\S*:\S*$/;
```

Comment:
This is over-inclusive. Synapse has a fairly well defined set of allowed characters for group IDs though: https://github.com/matrix-org/synapse/blob/2c3e995f38a411647ca08ae2959add577041ca97/synapse/types.py#L195-L214